### ISSUE TITLE
Possible BC break in `CacheItemPoolDecorator` since 2.10.2

### ISSUE THREAD

**Issue by boesing** [laminas/laminas-cache#99]
As per https://github.com/laminas/laminas-cache-storage-adapter-apcu/pull/7#issuecomment-831262536 I think this is a BC Break, take a look at:

https://github.com/php-cache/integration-tests/blob/824633c9547dc3c7ac5faeca03dd82939cdeb73c/src/CachePoolTest.php#L303

Missing keys shouldn't result in a `return false` here.

_Originally posted by @Slamdunk in https://github.com/laminas/laminas-cache/pull/85#discussion_r625089194_

**Comment by boesing** [comment]
TL;DR: `null` is never returned by any of the underlying cache backends. This means, that the comparison of `CacheItemPoolDecorator` is just useless and thus it returns `true` in **any** case, no matter if the keys were deleted or not.

---

Imho, that is a problem of the underlying storage.
`StorageInterface::removeItems` is clear on this:

https://github.com/laminas/laminas-cache/blob/15e0b44324a1dae500648ca78bd56e8671565661/src/Storage/StorageInterface.php#L196

I've recently found the same problem in `laminas-cache-storage-adapter-redis`.
The thing is, that it only worked due to the fact that the `CacheItemPoolDecorator` tested for an impossible return type. So actually, `true` was returned in any case. Even tho, deletion was not successful due to a cache backend error, e.g.

I am uncertain on how we should handle this, as there are adapters which actually do return an empty array.

What we could do is to pre-filter all keys, which do not exist. After that, we only pass those keys to `deleteItems` which might exist. This may ensure that the underlying storage does not return keys for those items which never existed in the first place.
This is how the `SimpleCacheDecorator` implemented that.

**Comment by boesing** [comment]
Released 2.10.3 and 2.11.1 with a fix.
